File: 5 - Form Template Method/Python/after/capital_strategy_revolver.py
from after.capital_strategy import CapitalStrategy


class CapitalStrategyRevolver(CapitalStrategy):
    def __init__(self):
        CapitalStrategy.__init__(self)

    def capital(self, loan):
        return super().capital(loan) + self.unused_capital(loan)

    def unused_capital(self, loan):
        return loan.unused_risk_amount() * self.duration(loan) * self.unused_risk_factor_for(loan)

    def risk_amount_for(self, loan):
        return loan.outstanding_risk_amount()

File: 4 - Replace Conditional Logic with Strategy/C#/MPG.ReplaceConditionalLogicWithStrategy/MPG.ReplaceConditionalLogicWithStrategy.After/CapitalStrategyRevolver.cs
namespace MPG.ReplaceConditionalLogicWithStrategy.After
{
    public class CapitalStrategyRevolver : CapitalStrategy
    {
        public override double Capital(Loan loan)
        {
            return (loan.OutstandingRiskAmount() * loan.Duration() * RiskFactor(loan))
                   + (loan.UnusedRiskAmount() * loan.Duration() * UnusedRiskFactor(loan));
        }
    }
}